db/migrate/20230323110408_add_changed_constraints_to_planning_application.rb

```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddChangedConstraintsToPlanningApplication < ActiveRecord::Migration[7.0]
+  def change
+    add_column :planning_applications, :changed_constraints, :text, array: true
```

Comment:
Can you explain why you've done it this way? I've not seen this kind of pattern before and it looks a bit strange to me to have this stored doubly, but I'm not sure!

Reply:
I based this on how `updated_address_or_boundary_geojson` is tracked and stored. In that case, a boolean is set if certain fields are changed; in this case I'm storing an array of the filenames — currently however we only care whether it's empty or not.

So I don't think it's storing doubly (unless I've misunderstood you) — at the moment, we store the new values when an application is modified, but we don't track the fact that they're different from the original values. This will track which values are updated, but not what they've changed from or to.

If we wanted to go a bit further, something like papertrail would be a lot more powerful, but that feels like overkill for now.

Reply:
Maybe a spec would help illustrate this?

i.e. make an update to constraints and then check what `planning_application.changed_constraints` is `[....]`. Then another update and check again etc.